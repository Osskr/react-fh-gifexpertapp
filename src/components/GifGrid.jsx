import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGif';
import GifGridItem from './GifGridItem';
//import { getGifs } from './helpers/getGifs';



const GifGrid = ({category}) => {


    const {data:images, loading} = useFetchGifs(category) //custom hook
      
    return (

        <div>
            <h3>{category}</h3>
         
            
            <div className='card-grid '>

                {loading ? <p className="animate__animated animate__bounce">Loading chaval</p> : null}
                { images.map((image)=> <GifGridItem key={image.id} 
                                                      {...image}/>)}
            </div> 
        </div>
        
    )
}

export default GifGrid
