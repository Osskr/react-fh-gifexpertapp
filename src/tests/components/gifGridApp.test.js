import React from 'react'
import {shallow} from 'enzyme'
import GifGrid from '../../components/gifGrid'

describe('Tests para el componente GifGridApp',()=>{

    test('debe mostrar GifGridApp correctamente', () => {

        const wrapper = shallow(<GifGrid/>)
        expect(wrapper).toMatchSnapshot()

    })

})