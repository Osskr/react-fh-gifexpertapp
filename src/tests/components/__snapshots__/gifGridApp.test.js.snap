// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests para el componente GifGridApp debe mostrar GifGridApp correctamente 1`] = `
<div>
  <h3 />
  <div
    className="card-grid "
  >
    <p
      className="animate__animated animate__bounce"
    >
      Loading chaval
    </p>
  </div>
</div>
`;
